### PR TITLE
AP_Scripting: raise max FLTE for yaw to 8

### PR DESCRIPTION
**Pull request by tridge**
better yaw for many vehicles

### CODE REVIEW

**Comment by timtuxworth** [comment]
> better yaw for many vehicles

Perhaps you could also update the .md file. it currently says "the YAW_FLTE filter will be set to a maximum of 2Hz".

**Review by Hwurzburg — DISMISSED** [review]
I don't think this actually works.....it will always be left at the default of 2.5 ....before it was always set to 2.0 (old FLTE_MAX) because the default exceeded it....now it does nothing

**Comment by tridge** [comment]
> I don't think this actually works.....it will always be left at the default of 2.5

this patch is just to stop quicktune from overriding and reducing FLTE so low. It allows the user to set a FLTE before tuning
Separately we could change the default away from 2.5

**Comment by tridge** [comment]
> Perhaps you could also update the .md file.

thanks! fixed

**Comment by Hwurzburg** [comment]
QuikTune does no tuning on Yaw FLTE at this point....it just resets it to FLTE_MAX if the tune starts with it either <=0 or greater than FLTE_MAX....I don't think you want to prevent someone from setting it > FLTE_MAX in some cases.....

having QuikTune either leave it alone totally or set it like you proposed at one time to GYRO_FLT/8 might be better?

**Comment by timtuxworth** [comment]
I was wondering if QuickTune should not tune yaw on a tilt rotor? From what I think I understand from what others have said (and have seen from my own  tuning), it doesn't seem to work on a tilt rotor. Not sure about other frame types.

File: libraries/AP_Scripting/applets/VTOL-quicktune.md [hunk 1]
```diff
@@ -151,11 +151,13 @@ With default settings the parameters to be tuned will be:
...
-The script will also adjust filter settings using the following rules:
+The script will also adjust filter settings using the following rules
+if QUIK_AUTO_FILTER is set to 1 (which is the default):
 
  - the FLTD and FLTT settings will be set to half of the INS_GYRO_FILTER value
- - the YAW_FLTE filter will be set to a maximum of 2Hz
- - if no SMAX is set for a rate controller than the SMAX will be set to 50Hz
+ - the YAW_FLTE filter will be set to a maximum of 8Hz
+
+Additionally, if no SMAX is set for a rate controller than the SMAX will be set to 50Hz.
```
Comment (timtuxworth) [inline libraries/AP_Scripting/applets/VTOL-quicktune.md:160]: then

**Comment by Hwurzburg** [comment]
> I was wondering if QuickTune should not tune yaw on a tilt rotor? From what I think I understand from what others have said (and have seen from my own tuning), it doesn't seem to work on a tilt rotor. Not sure about other frame types.

Andy Piper and I have had it tune yaw fine in two different vectored yaw QPlanes....just the FLTE was too low after....for me setting it to 10Hz fixed the issue after QuickTune

**Comment by andyp1per** [comment]
> Andy Piper and I have had it tune yaw fine in two different vectored yaw QPlanes....just the FLTE was too low after....for me setting it to 10Hz fixed the issue after QuickTune

I actually can no longer repeat this using such a high value, I had to use 2.5Hz. @lthall explained the basic problem to me - with vectored yaw and tri-tilts in particular there is a momentum effect from the movement of the motors backwards and forwards and the momentum effect is a physical D term working in the *wrong direction*. This makes it very hard to tune good values of D - at least for QAutotune - it's possible that QuickTune is doing better here since it is just looking for oscillation, but its also a very complex mechanical system so hard to see what the right approach might be. Using a low FLTE linearizes the effect, but then has other issues.

**Comment by timtuxworth** [comment]
As discussed in the dev call 2024-02-12, I flew this yesterday with Q_A_RAT_Y_FLTE = 10 and Q_A_ACCEL_Y_MAX = 4000 (reduced from 6000) and still getting a slow Yaw oscillation. Here are 3 logs:
https://www.dropbox.com/scl/fi/vc09k3hfx9n50hvejw2bx/log_85_2024-2-11-16-19-54.bin?rlkey=1vuky6rqprkrd2fghbcr0qshk&dl=0
https://www.dropbox.com/scl/fi/ewmefj662q5cgn789b50n/log_86_2024-2-11-16-40-08.bin?rlkey=3pl16nr8emvyc1d2dfwlgf4ew&dl=0
https://www.dropbox.com/scl/fi/i5uwv7mms8r273v2zeu0k/log_87_2024-2-11-16-58-18.bin?rlkey=eo82mrzopugk3szsefnefsfbj&dl=0

**Pull request merged**